packages/embark/src/lib/modules/contracts_manager/index.js

```diff
@@ -299,6 +302,24 @@ class ContractsManager {
           );
         callback(null, allContractsCompiled);
       },
+      function filterUnchangedContracts(allContractsCompiled, callback) {
+        let contractHashMap = fs.existsSync(CONTRACTS_HASHES_FILE) ?
+          require(CONTRACTS_HASHES_FILE) : {};
+        self.contractsFiles.filter(contractFile => contractFile.type !== Types.http).forEach(contractFile => {
```

Comment:
We filter out `Http` type files here because in order to generate a hash based on their source we have to wait until the source file has actually been fetched by embark, which happens much later in the process.

So for now we're ignoring those, resulting in always compiling them. This shouldn't be too much of a big deal as using http imports isn't a super common thing to do.